Makefile.common

```diff
@@ -147,6 +158,18 @@ $(PREFIX_O)%.o: %.c
 	$(SIL)$(CC) -c $(CFLAGS) "$(abspath $<)" -o "$@"
 	$(SIL)$(CC) -M  -MD -MP -MF $(PREFIX_O)$*.c.d -MT "$@" $(CFLAGS) $<
 
+$(PREFIX_O)%.o: %.cc
+	@mkdir -p $(@D)
+	@printf "CXX  %-24s\n" "$<"
+	$(SIL)$(CXX) -c $(CXXFLAGS) "$(abspath $<)" -o "$@"
+	$(SIL)$(CXX) -M  -MD -MP -MF $(PREFIX_O)$*.cc.d -MT "$@" $(CXXFLAGS) $<
+
+$(PREFIX_O)%.o: %.cpp
```

Comment:
Hmmm, can't we do just one rule `$(PREFIX_O)%.o: %.cc %.cpp`?

Reply:
Nah, we need to split/cleanup `FLAGS` depending on the language (it will be done at a later stage):
- `CPPFLAGS` should be flags for preprocessor
- `CFLAGS` should be flags for C-language compiler
- `CXXFLAGS` should be flags for C++-language compiler

Even now some CFLAGS are not accepted by `g++`, I assume there will be more in the future:
<img width="288" alt="image" src="https://user-images.githubusercontent.com/3939983/202699055-c260a9c7-ffec-48e6-90b0-17974e7be05b.png">